src/grpcbox_client.erl

```diff
@@ -80,22 +80,30 @@ unary_handler(Ctx, Channel, Path, Input, Def, Options) ->
                       service_def => Def},
                 case recv_end(S, 5000) of
                     eos ->
-                        {ok, Headers} = recv_headers(S, 0),
-                        case recv_trailers(S) of
-                            {ok, {<<"0">>, _, Metadata}} ->
-                                case recv_data(S, 0) of
-                                    {ok, Data} ->
-                                        {ok, Data, #{headers => Headers,
-                                                     trailers => Metadata}};
-                                    stream_finished ->
-                                        {ok, <<>>, #{headers => Headers,
-                                                     trailers => Metadata}}
-                                end;
-                            {ok, {Status, Message, Metadata}} ->
-                                {error, {Status, Message}, #{headers => Headers,
+                        case recv_headers(S, 0) of
+                            {ok, Headers} ->
+                                case recv_trailers(S) of
+                                    {ok, {<<"0">>, _, Metadata}} ->
+                                        case recv_data(S, 0) of
+                                            {ok, Data} ->
+                                                {ok, Data, #{headers => Headers,
                                                              trailers => Metadata}};
-                            timeout ->
-                                {error, timeout}
+                                            stream_finished ->
+                                                {ok, <<>>, #{headers => Headers,
+                                                             trailers => Metadata}}
+                                        end;
+                                    {ok, {Status, Message, Trailers}} ->
+                                        {error, {Status, Message}, #{headers => Headers,
+                                                                     trailers => Trailers}};
+                                    {error, _}=Error ->
+                                        Error
```

Comment:
we're willingly dropping timeout as a result here and crashing instead, right? Or would it make sense to keep the timeout since we now handle errors better?